src/x11/x.c

```diff
@@ -326,20 +326,14 @@ gboolean x_mainloop_fd_dispatch(GSource *source, GSourceFunc callback, gpointer
                                 wake_up();
                         }
                         break;
-                case FocusIn:
-                        LOG_D("XEvent: processing 'FocusIn'");
-                        wake_up();
-                        break;
-                case FocusOut:
-                        LOG_D("XEvent: processing 'FocusOut'");
-                        wake_up();
-                        break;
                 case CreateNotify:
                         LOG_D("XEvent: processing 'CreateNotify'");
                         if (win->visible &&
                             ev.xcreatewindow.override_redirect == 0)
                                 XRaiseWindow(xctx.dpy, win->xwin);
                         break;
+                case FocusIn:
+                case FocusOut:
```

Comment:
I don't get the point of removing the `FocusOut`/`FocusIn` cases. I don't get the point why we even covered these cases.

But at least, the following `LOG_D` call becomes misleading in this commit.

Reply:
According to my tests not checking for focus events breaks i3 when using the keyboard to switch between monitors. I'll chalk it up to "black magic" and leave it as is, that if statement stops draw from being called more than necessary anyway.

Reply:
Do you have a reproducer for this? I've tested now dunst with the cases `FocusIn`/`FocusOut` both removed. And it doesn't even make any difference on my system.

Because the window associated to the `Focus(In|Out)` event never matches our window and when receiving a FocusChange event, there's always another `PropertyNotify` event in the queue, which has an atom `_NET_ACTIVE_WINDOW`, which resembles exactly the `Focus(In|Out` event.

Reply:
Ouch. This hurts! I found the bug :see_no_evil: :angry: 

I guess, Captain Obvious hasn't called us, but we have to update the window's current screen more often: https://github.com/bebehei/dunst/commit/1faf7c671df379ec3c87778f92ac225cf7612d1e

And to debug the error, use this commit: https://github.com/bebehei/dunst/commit/a423c204e1524d2b57ca83f72fd1eede5b29337e

Look at the debug output while processing the events. The `get_screen_active()` when using the keyboard to switch has got a HUGE delay. And when listening to the `FocusChange` event, it does exactly that. Can you also see, that the window gets moved between the monitors with a longer delay, when using the keyboard?

Reply:
Yikes, where is that delay coming from though? I can still reproduce the issue with the commit that updates the active screen more often, the debug commit is the one that fixes it (specifically, the delay).